Fix travis gcc 4.7 build

Example failure: https://travis-ci.org/whoshuu/cpr/jobs/287726632

When compiling `server.cpp`, we see the error:

> /home/travis/build/whoshuu/cpr/test/server.cpp:68:5: error: ‘sleep_for’ is not a member of ‘std::this_thread’

--- a/.travis.yml
+++ b/.travis.yml
@@ -37,7 +37,7 @@ matrix:
           sources: *sources
 
     # gcc 4.7
-    - env: COMPILER=g++-4.7 CCOMPILER=gcc-4.7
+    - env: COMPILER=g++-4.7 CCOMPILER=gcc-4.7 EXTRA_CXXFLAGS=-D_GLIBCXX_USE_NANOSLEEP
       addons:
         apt:
           packages: g++-4.7
@@ -110,6 +110,7 @@ install:
 
 before_script:
   - export CXX=$COMPILER CC=$CCOMPILER
+  - export CXXFLAGS=$EXTRA_CXXFLAGS
   - cmake --version
   - mkdir build
   - cd build

